drivers/bmx055/bmx055_saul.c

```diff
@@ -24,7 +24,9 @@
 
 static int read_mag(const void *dev, phydat_t *res)
 {
-    if (bmx055_mag_read((bmx055_t *)dev, (int16_t *)res) != BMX055_OK) {
+    const bmx055_t *mydev = (const bmx055_t *)dev;
+
+    if (bmx055_mag_read(mydev, res->val) != BMX055_OK) {
```

Comment:
to me it would be even better to use `&res->val[0]` as you did in https://github.com/RIOT-OS/RIOT/pull/10711, that makes it explicit we want a single value, compare to gyros and mags that actually expect a triplet of `int16_t`s

Reply:
In this case `bmx055_mag_read` actually fills in all three elements of `res->val`
So let's keep it that way.

Reply:
ah damn, right, leave as is ... my bad (eyes)

Reply:
sorry for the noise